### PR TITLE
Fix typescript-node generation when only models are generated

### LINKED ISSUE #7126: [BUG] [typescript-node] Generating only models generates invalid TS
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?

##### Description

When providing `--global-property models` to generate only models, the resultant TypeScript references `../api` for importing `RequestFile`.

##### openapi-generator version

```❯ npx openapi-generator version
4.3.1
```

Not a regression that I know of.

##### OpenAPI declaration file content or url

Any valid OpenAPI file containing models will reproduce this.

```yaml
openapi: 3.0.2
info:
  version: 0.3.0
  description: null
  title: '@devsupply/core-types-typedefs'
paths:
  /foo:
    get:
      tags:
        - foo
      operationId: getFoo
      responses:
        "204":
          description: Foo
components:
  schemas:
    CloudProvider:
      type: string
      enum:
        - AWS
        - AZURE
        - GOOGLE
      x-enum-descriptions:
        - Amazon AWS
        - Microsoft Azure
        - Google Cloud Platform
```

##### Generation Details

```
npx openapi-generator generate 
    -g typescript-node 
    --global-property models 
    -i ./example.yml -o gen/
```

##### Steps to reproduce

Generate code as above. Attempt to build via TypeScript:

```
npx tsc gen/model/cloudProvider.ts
```
Results in:
```
13 import { RequestFile } from '../api';
                               ~~~~~~~~


Found 1 error.
```

##### Suggest a fix

Suggest moving the `RequestFile` definition into `models.ts` and also generating this file when `--global-property models` is provided. Will submit a PR with this fix...

Related Issue: #7040 (it should be noted that moving the `RequestFile` definition will not work until #7040 is addressed, as the validators import things from `request`)

### PR DESCRIPTION
**Pull request by sarumont**
- move export of all models to model/index.ts
- move RequestFile definition to model/index.ts

This fixes #7126

Manual testing has been performed with the minimal example provided in #7126. Current tests do not validate compilation, it seems (or #7126 would not exist). Please comment if such tests are desired.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### TODO:

- [x] add fallbacks to allow merging into a `minor`
- [x] move PR to the correct branch

### CODE REVIEW

**Comment by macjohnny** [comment]
@sarumont please copy the technical comittee as soon as this PR is ready for review

**Comment by sarumont** [comment]
CCI build is still failing on some unrelated templates being out of date, but this PR is ready for review otherwise.

CC @TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx @macjohnny @topce @akehir @petejohansonxo @amakhrov

**Comment by macjohnny** [comment]
@sarumont please merge the most recent master

**Comment by sarumont** [comment]
@macjohnny Merged and pushed.

File: samples/server/petstore/php-symfony/SymfonyBundle-php/.openapi-generator/FILES [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
-./Model/Pet.php
-./Model/Tag.php
-./Model/User.php
-./Service/JmsSerializer.php
-./Service/SerializerInterface.php
-./Service/StrictJsonDeserializationVisitor.php
-./Service/SymfonyValidator.php
-./Service/TypeMismatchException.php
-./Service/ValidatorInterface.php
-./Tests/AppKernel.php
```
Comment (macjohnny) [inline samples/server/petstore/php-symfony/SymfonyBundle-php/.openapi-generator/FILES:21]: please revert the changes in the `/FILES` files, as these cause the CI to fail.
unfortunately, I haven't investigated why this occurs.
cc @wing328

Comment (sarumont) [reply samples/server/petstore/php-symfony/SymfonyBundle-php/.openapi-generator/FILES]: Reverted and also updated to latest `master`

File: samples/client/petstore/typescript-angular-v10-provided-in-root/builds/with-npm/.openapi-generator/FILES [hunk 2]
```diff
@@ -9,12 +9,12 @@ configuration.ts
 encoder.ts
 git_push.sh
 index.ts
-model/./apiResponse.ts
```
Comment (macjohnny) [inline samples/client/petstore/typescript-angular-v10-provided-in-root/builds/with-npm/.openapi-generator/FILES:12]: I think all changed `*/FILES` in your PR should be reverted.
Please also merge the current master again, to fix the build

Comment (sarumont) [reply samples/client/petstore/typescript-angular-v10-provided-in-root/builds/with-npm/.openapi-generator/FILES:12]: Will do...those were all updated via `generate-samples.sh`, so there may be a bug there, too?

Comment (sarumont) [reply samples/client/petstore/typescript-angular-v10-provided-in-root/builds/with-npm/.openapi-generator/FILES]: It actually looks like `master` has the same changes for all `FILES` now, so hopefully this build will complete...

File: samples/client/petstore/typescript-node/default/model/index.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+export * from './apiResponse';
```
Comment (macjohnny) [inline samples/client/petstore/typescript-node/default/model/index.ts:1]: is this index file still being generated?
why does the RequestDetailedFile occur in index.ts and models.ts?

File: modules/openapi-generator/src/main/resources/typescript-node/models.mustache [hunk 4]
```diff
@@ -1,10 +1,27 @@
...
+    value: Buffer;
+    options?: {
+        filename?: string;
+        contentType?: string;
+    }
+}
+
+export type RequestFile = string | Buffer | fs.ReadStream | RequestDetailedFile;
+
+// Object serialization only relevant if generating APIs, too
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-node/models.mustache:23]:
```suggestion
{{! Object serialization only relevant if generating APIs, too }}
```
see https://mustache.github.io/mustache.5.html#Comments

File: samples/client/petstore/typescript-node/default/model/index.ts (hunk 3 above)
Comment (sarumont) [reply samples/client/petstore/typescript-node/default/model/index.ts]: Nope, it's not being generated anymore - removed!

**Review by macjohnny — APPROVED** [review]
LGTM

**Pull request merged**